Makefile

```diff
@@ -9,13 +9,17 @@ SYSROOT ?= $(CURDIR)/sysroot
 # A directory to install to for "make install".
 INSTALL_DIR ?= /usr/local
 # single or posix
-THREAD_MODEL = single
```

Comment:
There still seem to be a lot of non-LTO related changes here?   Some kind of re-base issue maybe?

Reply:
I've now filed #186 to pull out several refactorings.